alphagov/paas-cf
# [#154917736] Monitor Cloud Foundry internal certificates

## What

We already had a job called check-var-store-certs, but it was running only every 30 days and was sending an email alert.

Changes:
* Rename the job to check-certificates
* Run the job daily
* Fix the job that it should not alert if the vars-store file is empty (new dev environments)
* Add check for the ipsec CA cert
* Do not send an email
* Create a Datadog monitor to alert if the job fails or doesn't run for 2 days.

## How to review

1. Review and merge https://github.com/alphagov/paas-team-manual/pull/221 first.
1. Code review.
1. Run the `teams/main/pipelines/create-cloudfoundry/jobs/check-certificates` job.
1. Change temporarily the alert threshold, run the job and check if it triggers the Datadog monitor. 

Shortcut for testing:
* enable Datadog in paas-bootstrap
* run the create-bosh-concourse job
* set `ENABLE_DATADOG` manually to true in `datadog-terraform-apply` job
* run `make dev run_job JOB=post-deploy`

## Who can review

Not me.
